### DISCUSSION TITLE
FindMarkers and latent.variable with Assay object

### DISCUSSION THREAD

**Discussion by skilpinen** [satijalab/seurat#4098] [Q&A] [answered]
Hi,

Can someone instruct what is the best way to handle situation where I want to run FindMarkers so that I set groups to be compared as cells.1 and cells.2, but I would like to set latent.variable to something that is in meta.data of the Seurat object. 

However, FindMarkers method for Seurat object do not allow cells.1 and cells.2 arguments, but instead of ident.1 and ident.2 arguments. Then using FindMarkers for specific assay only with [['assay_name']] indexing from Seurat object allows me to set cells.1 and cells.2, but then it fails to find latent.variable data which doesn't seem to be present in this Assay anymore.

Obviously I can circumvent this problem by creating new ident class with cell groups I want to but it is really cumbersome to do. I assume there is smarter way which I haven't figured out?

**Comment by timoast** [comment, marked as answer]
Since cell metadata is stored at the Seurat object level, not in the Seurat assay, you will need to run FindMarkers on the Seurat object if you want to include a latent variable. Rather than specifying `cells.1` and `cells.2`, you should add a metadata variable that groups the cells and then use the `ident.1` and `ident.2` parameters in `FindMarkers`. You can easily add this using the `$<-` assignment operator if you already have vectors of cells, for example `object$group1 <- colnames(object) %in% cells`, then run `FindMarkers(object, ident.1 = TRUE, ident.2 = FALSE, group.by = "group.1", latent.vars = ...)`
